inference-engine/tests/functional/plugin/shared/src/single_layer_tests/non_max_suppression.cpp

```diff
@@ -0,0 +1,151 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "single_layer_tests/non_max_suppression.hpp"
+
+namespace LayerTestsDefinitions {
+
+using namespace ngraph;
+using namespace InferenceEngine;
+using namespace FuncTestUtils::PrecisionUtils;
+
+std::string NmsLayerTest::getTestCaseName(testing::TestParamInfo<NmsParams> obj) {
+    InputShapeParams inShapeParams;
+    InputPrecisions inPrecisions;
+    int32_t maxOutBoxesPerClass;
+    float iouThr, scoreThr, softNmsSigma;
+    op::v5::NonMaxSuppression::BoxEncodingType boxEncoding;
+    bool sortResDescend;
+    element::Type outType;
+    std::string targetDevice;
+    std::tie(inShapeParams, inPrecisions, maxOutBoxesPerClass, iouThr, scoreThr, softNmsSigma, boxEncoding, sortResDescend, outType, targetDevice) = obj.param;
+
+    size_t numBatches, numBoxes, numClasses;
+    std::tie(numBatches, numBoxes, numClasses) = inShapeParams;
+
+    Precision paramsPrec, maxBoxPrec, thrPrec;
+    std::tie(paramsPrec, maxBoxPrec, thrPrec) = inPrecisions;
+
+    std::ostringstream result;
+    result << "numBatches=" << numBatches << "_numBoxes=" << numBoxes << "_numClasses=" << numClasses << "_";
+    result << "paramsPrec=" << paramsPrec << "_maxBoxPrec=" << maxBoxPrec << "_thrPrec=" << thrPrec << "_";
+    result << "maxOutBoxesPerClass=" << maxOutBoxesPerClass << "_";
+    result << "iouThr=" << iouThr << "_scoreThr=" << scoreThr << "_softNmsSigma=" << softNmsSigma << "_";
+    result << "boxEncoding=" << boxEncoding << "_sortResDescend=" << sortResDescend << "_outType=" << outType << "_";
+    result << "TargetDevice=" << targetDevice;
+    return result.str();
+}
+
+void NmsLayerTest::ConfigureNetwork() {
+    int i = 0;
+    for (const auto &out : cnnNetwork.getOutputsInfo()) {
```

Comment:
IMO, using `for (size_t i = 0; i <cnnNetwork.getOutputsInfo().size(); i++) {` is more clear

Reply:
I think it is your personal preference 😄 

`foreach` construction is easier.

Reply:
And increasing `i` in cycle? XD So interesting)

Reply:
fixed